source/release-notes/1.6.txt

```diff
@@ -0,0 +1,145 @@
+
+- Changed telemetry to also send error messages to Segment when ``mongosync``
+  exits with a fatal error.
+
+- Added support for ``mongosync`` to sync between source and destination 
+  clusters that have the same major and minor versions, but different 
+  patch versions.
+
+- Upgraded build to use Go 1.20.
+
+- ``mongosync`` now periodically logs when server operations take a longer 
+  than expected time.
+
+Issues Fixed:
+
+- Fixed a bug introduced in 1.5, where ``mongosync`` could fail to apply change
+  events for a document if they occur shortly after the document is copied
+  during the Collection Copy phase.
+
+- Fixed a bug where ``mongosync`` could crash due to the source cluster
+  containing legacy indexes with unknown index options.
+
+- Fixed a bug where ``mongosync`` could crash with an error during change event
+  application, if the application fails to read 500 documents within 5 minutes
+  or reaches the end of the oplog when reading from change streams on the
+  source cluster.
+
+- Fixed a bug where ``mongosync`` could crash when indexes apply constraints to
+  collection documents.
+
+  This crash occured when such an index was added or dropped at nearly the same
+  time as an index that violated these constrains was deleted or inserted.  It
+  would manifest with an error like:
+
+  .. code-block:: text
+     :copyable: false
+
+     Ambiguous field name found in array (do not use numeric field names in embedded elements in an array)
+
+  or a similar error. This fix makes this crash less likely, but it does not
+  elimiate the possibility that it will occur.
+
+- Fixed a bug where ``mongosync`` can fail to copy certain documents if the
+  insertion of documents causes duplicate key errors at the same time as a
+  write concern error. This bug has been present in mongosync since its first
+  release and may have arisen if the destination cluster logs reported a write
+  concern error alongside a duplicate key error.
+
+- Fixed a bug introduced in ``mongosync`` 1.5 where change events may be
+  skipped during the Collection Copy phase when there is little write activity
+  on the source cluster.
+
+- Fixed a bug introduced in 1.5 where ``mongosync`` could cause data
+  consistency errors in collections with non-default collation. 
+
+  This issue only affects collections with non-simple collation that are present
+  during ``mongosync`` initialization. Collections with non-simple collation
+  created after ``mongosync`` initialization are not affected.
+
+  .. note::
+
+     Users who used Oplog Rollover Resilience with at least one collection with
+     non-simple collation and string _id fields may have been affected by this
+     issue. 
+     
+     In order to mitigate this issue, ``monogsync`` 1.6.0 automatically
+     disables Oplog Rollover Resilience if it finds at least one collection with
+     non-simple collation during initialization. We will investigate a
+     more permanent solution for future releases.
```

Comment:
The team would like to change this bullet to:

> Fixed a bug introduced in Mongosync v1.5.0 where Mongosync could cause data consistency errors in collections with non-default collation. This issue only affects collections with non-simple collation that are present during Mongosync initialization and contain documents with string `_id` values. Any collections with non-simple collation created after Mongosync initialization or whose documents have all non-string `_id` values are unaffected.
> 
> In order to mitigate this issue, Mongosync v1.6.0 will automatically disable Oplog Rollover Resilience if it finds at least one collection with non-simple collation during Mongosync initialization. We will investigate a more permanent solution for future releases.

(Feel free to put some of this in a `.. note::` if you think that's appropriate.)

Reply:
Fixed